crates/bevy_ecs/src/removal_detection.rs

```diff
@@ -249,7 +271,7 @@ where
     type Item = Entity;
     type IntoIter = RemovedIter<'a>;
     fn into_iter(self) -> Self::IntoIter {
-        self.iter()
```

Comment:
We should probably remove the `IntoIterator` implementation, like #9583, although it doesn't need to be done in this PR.

Reply:
Should I create another pull request for it or can I commit it on this pull request ?

Reply:
Add it to this PR.